tools/jsonschemavalidator/main.cpp

```diff
@@ -0,0 +1,204 @@
+/*=============================================================================
+
+Library: CppMicroServices
+
+Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+file at the top-level directory of this distribution and at
+https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=============================================================================*/
+
+#include "optionparser.h"
+#include "rapidjson/error/en.h"
+#include "rapidjson/filereadstream.h"
+#include "rapidjson/schema.h"
+#include "rapidjson/stringbuffer.h"
+#include "rapidjson/prettywriter.h"
+#include <memory>
+#include <iostream>
+#include <string>
+
+using namespace rapidjson;
+
+// functor object for deleting FILE objects
+struct FILEDeleter
+{
+  typedef FILE* pointer;
+  void operator()(FILE *pFile)
+  {
+    if (pFile)
+      fclose(pFile);
+  }
+};
+
+/**
+ * \brief Utility function to create a rapidjson::Document object from a file path
+ *
+ * \jsonfile is the file path of the json data
+ *
+ * \returns rapidjson::Document object representing the json file
+ * \throws std::runtime_error if the input file is non-existent or is not a valid json file
+ */
+Document ReadJSONDocument(const std::string& filePath)
+{
+  std::unique_ptr<FILE, FILEDeleter> fp(fopen(filePath.c_str(), "r"));
+  if (!fp)
+  {
+    std::string err("File '" + filePath + "' not found");
+    throw std::runtime_error(err);
+  }
+  char buffer[4096];
```

Comment:
should the json document be limited to 4096 bytes? Can [BasicIStreamWrapper](http://rapidjson.org/classrapidjson_1_1_basic_i_stream_wrapper.html) be used?

Reply:
What Jeff said :-)